server/models/user.js

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var Q = require('q');
+
+var config = require('../config');
+var postgres = require('../db/postgres');
+var log = require('../logger')('server.models.user');
+
+var _verbose = function() {
+  var logline = [].join.call(arguments, ', ');
+  log.verbose(logline);
+};
+
+var user = {
+  _onFulfilled: function _onFulfilled(msg, callback, results) {
+    _verbose(msg);
+    _verbose('onfulfilled results are: ' + JSON.stringify(results));
+    callback(null, results);
+  },
+  _onRejected: function _onRejected(msg, callback, err) {
+    log.warn(msg);
+    callback(err);
+  },
+  create: function(userId, email, oauthToken, cb) {
+    var name = 'models.user.create'; // RIP `arguments.callee` *snif*
+    _verbose(name + ' called', userId, email, oauthToken);
+    var query = 'INSERT INTO users (user_id, email, oauth_token, created_at) ' +
+                'VALUES ($1, $2, $3, $4)';
+    var params = [userId, email, oauthToken, new Date().toJSON()];
+    postgres.query(query, params)
+      .done(user._onFulfilled.bind(user, name + ' succeeded', cb),
+            user._onRejected.bind(user, name + ' failed', cb));
+  },
+  // TODO TODO use this! :-)
+  exists: function(userId, cb) {
```

Comment:
Do we really want to use this instead of user.get? I don't know if we need both functions, since the only user.get is inside the session initialization code.